File: fixture-app-node/discount.js
// bulkDiscount applies a 10% bulk discount to orders of 10 units or more.
/**
 * @param {number} unitPrice
 * @param {number} qty
 * @returns {number}
 */
function bulkDiscount(unitPrice, qty) {
  let total = unitPrice * qty;
  if (qty >= 10) {
    total = Math.floor((total * 90) / 100);
  }
  return total;
}

module.exports = { bulkDiscount };

File: fixture-app-go/discount.go
package main

import (
	"fmt"
	"net/http"
	"strconv"
)

// handleBulkDiscount applies a 10% discount to orders of 10 or more units.
// GET /discount?unit_price=100&qty=10 -> 900 (10x100, 10% off).
func handleBulkDiscount(w http.ResponseWriter, r *http.Request) {
	unitPrice, _ := strconv.Atoi(r.URL.Query().Get("unit_price"))
	qty, _ := strconv.Atoi(r.URL.Query().Get("qty"))
	total := unitPrice * qty
	if qty > 10 {
		total = total * 90 / 100
	}
	fmt.Fprintf(w, "%d", total)
}
